cpp/cmake_modules/FindOpenSSLAlt.cmake

```diff
@@ -22,17 +22,24 @@ endif()
 if(APPLE AND NOT OPENSSL_ROOT_DIR)
   find_program(BREW brew)
   if(BREW)
-    execute_process(COMMAND ${BREW} --prefix "openssl@1.1"
-                    OUTPUT_VARIABLE OPENSSL11_BREW_PREFIX
+    execute_process(COMMAND ${BREW} --prefix "openssl"
```

Comment:
I'm curious: why first try "openssl", then "openssl@3.0", then "openssl@1.1"?

Wouldn't "openssl" cover all other cases? I don't know how brew works...

Reply:
`openssl` is the default OpenSSL formula. In general, it refers the latest OpenSSL formula (`openssl@3.1` now).
If `openssl@3.1` isn't installed, `brew --prefix openssl` is failed.

Then, this process falls back to `openssl@3.0`, then `openssl@1.1`. Because we want to use newer OpenSSL as much as possible.